### ISSUE TITLE
ME1 Savegame serialization issue

### ISSUE THREAD

**Issue by Mgamerz** [electronicarts/MELE_ModdingSupport#3]
I'm working on converting this to C# classes and am seeing a duplicate variable declared here of `ClassName`: 
https://github.com/electronicarts/MELE_ModdingSupport/blob/master/DataFormats/ME1_Savegame.h#L377

I assume this is an accident?

**Comment by KarlitosVII** [comment]
It's not a class name like Soldier or Adept but an Unreal class name needed by the game when it deserialize the file.
In your example this is the class name of the squad and the inventory but they are only deserialized if the boolean is true.

![Capture](https://user-images.githubusercontent.com/7722685/230751650-55dc022a-07f1-4ea9-8888-d7898919bbfb.PNG)

**Comment by Mgamerz** [comment]
Yes but shouldn't this have a unique name? This would not compile in C++
either.

**Comment by KarlitosVII** [comment]
It's just a string that tells the deserializer how to deserialize the next bytes. You have to read this string to know which class to use. Just like I did here:

https://github.com/KarlitosVII/trilogy-save-editor/blob/main/src/save_data/mass_effect_1_le/legacy/mod.rs#L62-L84

In the same way, when you serialize you must indicate the name of the class before serializing the object.

Throughout the `Legacy` part of LE1, as soon as you see `ClassName` and `LEGACY_BaseObjectSaveRecord `, the only way to guess how to deserialize is at runtime.
